action.yml

```
@@ -33,9 +41,64 @@ runs:
           exit 1
         fi
 
-        curl -L https://github.com/rudderlabs/rudder-iac/releases/download/${{ inputs.cli_version }}/rudder-cli_Linux_x86_64.tar.gz | tar -xz rudder-cli
-        chmod +x rudder-cli
-        sudo mv rudder-cli /usr/local/bin/
+        echo "Downloading Rudder CLI ${{ inputs.cli_version }}..."
+        
+        # Create cache directory
+        mkdir -p ~/.local/bin
+        
```

Should we add these defensive checks ?
I mean we already would have enough signals to know something went wrong right ? 
So we won't need these many logging statements IMO.


```suggestion
        
        curl -L https://github.com/rudderlabs/rudder-iac/releases/download/${{ inputs.cli_version }}/rudder-cli_Linux_x86_64.tar.gz | tar -xz -C ~/.local/bin rudder-cli
        chmod +x ~/.local/bin/rudder-cli
        ~/.local/bin/rudder-cli --version
        
        echo "Rudder CLI downloaded and cached successfully"
```

---

since this action is user facing i want it to be as informative as possible..it will also help us in debugging as the user would be able to give us exact errors